### PR TITLE
Bump beam to 2.61, pin beam dependencies, reduce dependabot noise

### PR DESCRIPTION
**Pull request by kellen**
Note that this involves downgrading some libraries to be in sync with the beam release

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -100,138 +100,145 @@
...
     <!-- https://github.com/googleapis/java-cloud-bom/releases -->
     <!-- https://storage.googleapis.com/cloud-opensource-java-dashboard/com.google.cloud/libraries-bom/26.45.0/index.html -->
     <google-cloud-libraries-bom.version>26.45.0</google-cloud-libraries-bom.version>
-    <guava.version>33.3.0-jre</guava.version>
-    <hamcrest.version>3.0</hamcrest.version>
+    <!-- versions from GCP bom -->
+    <opencensus.version>0.31.1</opencensus.version>
+
+    <!-- other deps -->
+    <avro.version>1.11.4</avro.version>
```
Comment (RustedBones) [inline pom.xml:134]: Shouldn't we put this on on the beam BOM? `2.60.0` uses `1.11.3`

File: pom.xml [hunk 2]
```diff
@@ -554,6 +543,31 @@
                     <!-- Keep aligned with prerequisite section below. -->
                     <version>[3.3.9,)</version>
                   </requireMavenVersion>
+                  <requireUpperBoundDeps>
+                    <excludes>
+                      <!-- managed by beam BOM -->
```
Comment (RustedBones) [inline pom.xml:548]: add avro here too, otherwise we'lle get undesired `1.12.0` update

File: pom.xml (hunk 1 above)
Comment (kellen) [reply pom.xml]: Yes, done

File: pom.xml [hunk 3]
```diff
@@ -100,138 +100,150 @@
...
+    <commons-codec.version>1.17.1</commons-codec.version> <!-- libraries-bom version -->
+    <commons-compress.version>1.26.2</commons-compress.version>
+    <errorprone.version>2.10.0</errorprone.version>
+    <guava.version>33.1.0-jre</guava.version>
+    <hamcrest.version>2.1</hamcrest.version>
+    <httpclient.version>4.5.13</httpclient.version>
+    <httpcore.version>4.4.14</httpcore.version>
+    <jackson.version>2.15.4</jackson.version>
+    <joda-time.version>2.10.14</joda-time.version>
+    <netty.version>4.1.100.Final</netty.version>
```
Comment (labianchin) [inline pom.xml:122]:
```suggestion
    <netty.version>4.1.115.Final</netty.version>
```

**Review by labianchin — COMMENTED** [review]
Reducing dependabot noise looks good.
Still, can you expand on why are dependencies being downgraded? Or why syncing with with older versions Beam release?

In my experience Beam does not keep dependencies up to date very well, sometimes carrying vulnerabilities. The up to date versions here have been "more compatible" with maven enforcer, and had no runtime issues (mostly minor bumps, no breaking changes).

File: pom.xml [hunk 4]
```diff
@@ -100,138 +100,150 @@
...
+    <slf4j.version>1.7.30</slf4j.version>
+    <threetenbp.version>1.6.8</threetenbp.version>
+    <zstd-jni.version>1.5.6-3</zstd-jni.version>
```
[the remark this answers is not in the document]
Comment (RustedBones) [reply pom.xml:125]: See above

File: pom.xml [hunk 5]
```diff
@@ -100,138 +100,150 @@
...
+
+    <!-- GCP BOM -->
     <!-- https://github.com/googleapis/java-cloud-bom/releases -->
     <!-- https://storage.googleapis.com/cloud-opensource-java-dashboard/com.google.cloud/libraries-bom/26.45.0/index.html -->
     <google-cloud-libraries-bom.version>26.45.0</google-cloud-libraries-bom.version>
```
[the remark this answers is not in the document]
Comment (RustedBones) [reply pom.xml:130]: We're syncing with beam's version

File: pom.xml [hunk 6]
```diff
@@ -554,6 +543,32 @@
                     <!-- Keep aligned with prerequisite section below. -->
                     <version>[3.3.9,)</version>
                   </requireMavenVersion>
+                  <requireUpperBoundDeps>
+                    <excludes>
+                      <!-- managed by beam BOM -->
```
[the remark this answers is not in the document]
Comment (RustedBones) [reply pom.xml:548]: This is the strategy we used with scio. We upgrade those libs along with beam

**Review by RustedBones — APPROVED** [review]

**Pull request merged**